src/doc/rustdoc/src/how-to-write-documentation.md

```diff
@@ -254,6 +254,134 @@ characters:
 
 So, no need to manually enter those Unicode characters!
 
+### Inline HTML
+
+As a standard Commonmark parser with no special restrictions, rustdoc allows
+you to write HTML whenever the regular markup isn't sufficient, such as
+advanced table layouts:
+
+```html
+<table>
+    <thead>
+        <tr>
+            <th>Name</th>
+            <th>Description</th>
+            <th>Category</th>
+        </tr>
+    </thead>
+    <tbody>
+        <tr>
+            <td>traits</td>
+            <td>static and dynamic dispatch</td>
+            <td>generics</td>
+        </tr>
+        <tr>
+            <td colspan="3">enums</td>
+        </tr>
+        <tr>
+            <td rowspan="2">doc comments</td>
+            <td colspan="2">reference documentation</td>
+        </tr>
+        <tr>
+            <td>markdown comments</td>
+            <td>attributes with custom syntax</td>
+        </tr>
+    </tbody>
+</table>
+```
+
+This will render the same way Markdown tables do, even though Markdown tables
+don't support `colspan`:
+
+<table>
+    <thead>
+        <tr>
+            <th>Name</th>
+            <th>Description</th>
+            <th>Category</th>
+        </tr>
+    </thead>
+    <tbody>
+        <tr>
+            <td>traits</td>
+            <td>static and dynamic dispatch</td>
+            <td>generics</td>
+        </tr>
+        <tr>
+            <td colspan="3">enums</td>
+        </tr>
+        <tr>
+            <td rowspan="2">doc comments</td>
+            <td colspan="2">reference documentation</td>
+        </tr>
+        <tr>
+            <td>markdown comments</td>
+            <td>attributes with custom syntax</td>
+        </tr>
+    </tbody>
+</table>
+
+HTML is not sanitized when included in the documentation, though
+improperly-nested tags will produce a build-time warning. It can be turned
+into an error by adding `#![deny(rustdoc::invalid_html_tags)]` to your crate.
```

Comment:
I don't think we need to mention `deny` here, it's implied by it being a lint and the `note` in the error message. I don't feel strongly, though.